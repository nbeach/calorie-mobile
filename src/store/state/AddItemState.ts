export interface AddItemState {
    readonly name: string
    readonly calories: string
}
