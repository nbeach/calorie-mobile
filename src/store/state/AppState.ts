import {ItemState} from "./ItemState"
import {AddItemState} from "./AddItemState"

export interface AppState {
    readonly items: ReadonlyArray<ItemState>,
    readonly addItem: AddItemState
}

