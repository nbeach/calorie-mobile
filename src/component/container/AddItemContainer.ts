import {connect, MapDispatchToPropsFunction, MapStateToProps} from "react-redux"
import {AppState} from "../../store/state/AppState"
import {AddItem, AddItemDispatchProps, AddItemStateProps} from "../presentation/AddItem"
import {addItemAction} from "../../store/action/AddItemAction"
import {addItemNameChangedAction} from "../../store/action/AddItemNameChangedAction"
import {addItemCaloriesChangedAction} from "../../store/action/AddItemCaloriesChangedAction";

export const mapStateToProps: MapStateToProps<AddItemStateProps, void, AppState> = (state) => ({
    ...state.addItem,
})

export const mapDispatchToProps: MapDispatchToPropsFunction<AddItemDispatchProps, void> = (dispatch) => ({
    addItem: (name: string, calories: number) => dispatch(addItemAction({name, calories})),
    itemNameChanged: (name: string) => dispatch(addItemNameChangedAction({name})),
    itemCaloriesChanged: (calories: string) => dispatch(addItemCaloriesChangedAction({calories})),
})

export const AddItemContainer = connect(mapStateToProps, mapDispatchToProps)(AddItem)
