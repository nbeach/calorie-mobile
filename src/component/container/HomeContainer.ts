import {connect, MapStateToProps} from "react-redux"
import {Home, HomeStateProps} from "../presentation/Home"
import {AppState} from "../../store/state/AppState"

export const mapStateToProps: MapStateToProps<HomeStateProps, void, AppState>  = (state) => state

export const HomeContainer = connect(mapStateToProps)(Home)
