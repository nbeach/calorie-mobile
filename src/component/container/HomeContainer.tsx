import {connect} from "react-redux";
import {Home, HomeProps} from "../presentation/Home";
import {AppState} from "../../store/state/AppState";

export const mapStateToProps = (state: AppState): HomeProps => state

export const HomeContainer = connect(mapStateToProps)(Home)

