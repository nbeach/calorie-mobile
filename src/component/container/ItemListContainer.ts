import {connect, MapDispatchToPropsFunction, MapStateToProps} from "react-redux"
import {AppState} from "../../store/state/AppState"
import {ItemList, ItemListDispatchProps, ItemListProps} from "../presentation/ItemList"
import {removeItemAction} from "../../store/action/RemoveItemAction"

export const mapStateToProps: MapStateToProps<ItemListProps, void, AppState> = (state) => ({
    items: state.items,
})

export const mapDispatchToProps: MapDispatchToPropsFunction<ItemListDispatchProps, void> = (dispatch) => ({
    removeItem: (id: string) => dispatch(removeItemAction({ id })),
})

export const ItemListContainer = connect(mapStateToProps, mapDispatchToProps)(ItemList)
