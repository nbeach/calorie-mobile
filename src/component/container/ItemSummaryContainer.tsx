import {ItemListDispatchProps} from "../presentation/ItemList"
import {connect, MapDispatchToPropsFunction, MapStateToProps} from "react-redux"
import {AppState} from "../../store/state/AppState"
import {removeItemAction} from "../../store/action/RemoveItemAction"
import {ItemSummary, ItemSummaryStateProps} from "../presentation/ItemSummary"

export const mapStateToProps: MapStateToProps<ItemSummaryStateProps, void, AppState> = ({totalCalories, items}) => ({ items, totalCalories })

export const mapDispatchToProps: MapDispatchToPropsFunction<ItemListDispatchProps, void> = (dispatch) => ({
    removeItem: (id: string) => dispatch(removeItemAction({ id })),
})

export const ItemSummaryContainer = connect(mapStateToProps, mapDispatchToProps)(ItemSummary)
