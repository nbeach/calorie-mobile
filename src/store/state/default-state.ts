import {AppState} from "./AppState"

export const DEFAULT_STATE: AppState = {
    items: [
        {name: "Carrots", calories: 1000},
        {name: "Chicken", calories: 500},
    ],
    addItem: {
        name: "",
        calories: "",
    },
}

