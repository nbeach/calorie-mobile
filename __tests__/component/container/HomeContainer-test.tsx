import React from "react"

import renderer from "react-test-renderer"
import {Provider} from "react-redux"
import {AppState} from "../../../src/store/state/AppState"
import {Store} from "redux"
import {HomeContainer} from "../../../src/component/container/HomeContainer"

describe(HomeContainer.name, () => {

    it("displays the items and allows adding items", () => {
        const state: AppState = {
            addItem:  {name: "Beef", calories: 200},
            items: [{name: "Chicken", calories: 100}],
        }

        const store: Store<AppState> = {
            getState: () => state,
            dispatch: jest.fn(),
            subscribe: jest.fn(),
            replaceReducer: jest.fn(),
        }

        const tree = renderer.create(
            <Provider store={store}>
                <HomeContainer state />
            </Provider>,
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

})

