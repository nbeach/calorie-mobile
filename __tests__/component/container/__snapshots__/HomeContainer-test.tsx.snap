// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect displays the items and allows adding items 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#e4ebee",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Item"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#AAAAAA",
          "borderStyle": "solid",
          "fontSize": 18,
          "width": "60%",
        }
      }
      testID="NameField"
      underlineColorAndroid="transparent"
      value="Beef"
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Cals"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#AAAAAA",
          "borderStyle": "solid",
          "fontSize": 18,
          "width": "15%",
        }
      }
      testID="CalorieField"
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1b5fff",
          "borderColor": "#EEEEEE",
          "borderStyle": "solid",
          "borderWidth": 1,
          "fontSize": 18,
          "justifyContent": "center",
          "opacity": 1,
          "width": "15%",
        }
      }
      testID="AddButton"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        +
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#AAAAAA",
          "borderStyle": "solid",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Chicken
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        100
      </Text>
    </View>
  </View>
</View>
`;
